From d7b5b0a30bc096ec7a6802a2806258f9e6e39fea Mon Sep 17 00:00:00 2001
From: Dongxu Li <dongxuli2011@gmail.com>
Date: Fri, 7 Jul 2023 07:44:29 -0400
Subject: [PATCH] Fixes for CodeQL scan (issue #1646)

47 issues fixed:

RS_Debug::print formating type/length mismatch (TODO: RS_Debug should support ostream type output);
Loop index comparison range mismatch: uint8_t vs uint32_t.

Squashed commit of the following:

commit 97f5c1724ea59fefaae73e6d13c8486101302faa
Author: Dongxu Li <dongxuli2011@gmail.com>
Date:   Fri Jul 7 07:35:32 2023 -0400

    More CodeQL fixes

commit 87d84e40539a89866cd3aee9aa9f5ed3fcb8699a
Author: Dongxu Li <dongxuli2011@gmail.com>
Date:   Fri Jul 7 01:47:18 2023 -0400

    More CodeQL fixes

commit 7d0e71bc820e2f32d04d673bae781a2348d0bbb6
Author: Dongxu Li <dongxuli2011@gmail.com>
Date:   Fri Jul 7 00:52:49 2023 -0400

    More CodeQL fixes

commit 4ce44e991617332d97d4a6d7726d9ef71adcbad5
Author: Dongxu Li <dongxuli2011@gmail.com>
Date:   Thu Jul 6 23:58:09 2023 -0400

    CodeQL scan fixes

    1. RS_Debug::print() type mismatch (TODO: support ostream style in
    RS_Debug);
    2. Range mismatch in unsigned comparison (uint8_t vs uint32_t)
---
 libraries/libdxfrw/src/drw_entities.cpp       |  4 +--

diff --git a/libraries/libdxfrw/src/drw_entities.cpp b/libraries/libdxfrw/src/drw_entities.cpp
index 625f393eba..165fe9e0dc 100644
--- a/src/drw_entities.cpp
+++ b/src/drw_entities.cpp
@@ -1105,7 +1105,7 @@ bool DRW_Insert::parseDwg(DRW::Version version, dwgBuffer *buf, duint32 bs){
             attH = buf->getHandle(); /* H 2 BLOCK HEADER (hard pointer) */
             DRW_DBG("second attrib Handle: "); DRW_DBGHL(attH.code, attH.size, attH.ref); DRW_DBG("\n");
         } else {
-            for (duint8 i=0; i< objCount; ++i){
+            for (dint32 i=0; i< objCount; ++i){
                 dwgHandle attH = buf->getHandle(); /* H 2 BLOCK HEADER (hard pointer) */
                 DRW_DBG("attrib Handle #"); DRW_DBG(i); DRW_DBG(": "); DRW_DBGHL(attH.code, attH.size, attH.ref); DRW_DBG("\n");
             }
@@ -2987,7 +2987,7 @@ bool DRW_Viewport::parseDwg(DRW::Version version, dwgBuffer *buf, duint32 bs){
         DRW_DBG("ViewPort ent header: "); DRW_DBGHL(someHdl.code, someHdl.size, someHdl.ref); DRW_DBG("\n");
     }
     if (version > DRW::AC1014) {//2000+
-        for (duint8 i=0; i < frozenLyCount; ++i){
+        for (duint32 i=0; i < frozenLyCount; ++i){
             someHdl = buf->getHandle();
             DRW_DBG("Frozen layer handle "); DRW_DBG(i); DRW_DBG(": "); DRW_DBGHL(someHdl.code, someHdl.size, someHdl.ref); DRW_DBG("\n");
         }
